mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala

```diff
@@ -171,56 +171,47 @@ object AlignmentRecordMaterialization extends Logging {
-          var alignments: AlignmentRecordRDD = null
-          // hack to get around issue in hadoop_bam, which throws error if contigName is not found in bam file
-          val path = new Path(fp)
-          val fileSd = SequenceDictionary(SAMHeaderReader.readSAMHeaderFrom(path, sc.hadoopConfiguration))
-          for (r <- List(regions._1, regions._2)) {
-            if (fileSd.containsRefName(r.referenceName)) {
-              val x = sc.loadIndexedBam(fp, r)
-              if (alignments == null) alignments = x
-              else alignments = alignments.transform(rdd => rdd.union(x.rdd))
-            }
-          }
-          alignments
-        case _ => sc.loadBam(fp)
+      if (regions.isDefined) {
+        // hack to get around issue in hadoop_bam, which throws error if contigName is not found in bam file
+        val path = new Path(fp)
+        val fileSd = SequenceDictionary(SAMHeaderReader.readSAMHeaderFrom(path, sc.hadoopConfiguration))
+        val predicateRegions: Iterable[ReferenceRegion] = regions.get
+          .flatMap(r => {
+            LazyMaterialization.getContigPredicate(r)
+          }).filter(r => fileSd.containsRefName(r.referenceName))
+        sc.loadIndexedBam(fp, predicateRegions)
+      } else {
+        sc.loadBam(fp)
       }
     }
   }
 
   /**
    * Loads ADAM data using predicate pushdowns
    * @param sc SparkContext
-   * @param region Region to load
+   * @param regions Iterable of ReferenceRegions to load
    * @param fp filepath to load from
    * @return RDD of data from the file over specified ReferenceRegion
    */
-  def loadAdam(sc: SparkContext, fp: String, region: Option[ReferenceRegion]): AlignmentRecordRDD = {
-    AlignmentTimers.loadADAMData.time {
-      val pred: Option[FilterPredicate] =
-        region match {
-          case Some(_) => {
-            val contigs = LazyMaterialization.getContigPredicate(region.get)
-            Some((LongColumn("end") >= region.get.start) && (LongColumn("start") <= region.get.end) &&
-              (BinaryColumn("contigName") === Binary.fromString(contigs._1.referenceName) ||
-                BinaryColumn("contigName") === Binary.fromString(contigs._2.referenceName)) &&
-                (BooleanColumn("readMapped") === true))
-          } case None => None
-        }
-      val proj = Projection(AlignmentRecordField.contigName, AlignmentRecordField.mapq, AlignmentRecordField.readName,
-        AlignmentRecordField.start, AlignmentRecordField.readMapped, AlignmentRecordField.recordGroupName,
-        AlignmentRecordField.end, AlignmentRecordField.sequence, AlignmentRecordField.cigar, AlignmentRecordField.readNegativeStrand,
-        AlignmentRecordField.readPaired, AlignmentRecordField.recordGroupSample)
-      sc.loadParquetAlignments(fp, predicate = pred, projection = Some(proj))
-    }
+  def loadAdam(sc: SparkContext, fp: String, regions: Option[Iterable[ReferenceRegion]]): AlignmentRecordRDD = {
+    val pred =
```

Comment:
Do we still want to call AlignmentTimers.loadADAMData.time here? Otherwise we don't use the loadADAMData timer at all.